boards/riscv/tlsr9518adk80d/tlsr9518adk80d_pinctrl.dtsi

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2022 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dt-bindings/pinctrl/b91-pinctrl.h>
+
+&pinctrl {
+	/* Enable PINCTRL driver */
+	status = "okay";
+
+	/* Set pad-mul-sel register value.
+	 * Note: Pins functions below (pinmux = <...>) depend on this value.
+	 */
+	pad-mul-sel = <1>;
+
+	/* UART0: TX(PA3 PB2 PD2), RX(PA4 PB3 PD3) */
+
+	uart0_tx_pa3: uart0_tx_pa3 {
+		pinmux = <B91_PINMUX_SET(B91_PORT_A, B91_PIN_3, B91_FUNC_B)>;
+	};
+	uart0_tx_pb2: uart0_tx_pb2 {
+		pinmux = <B91_PINMUX_SET(B91_PORT_B, B91_PIN_2, B91_FUNC_C)>;
+	};
+	uart0_tx_pd2: uart0_tx_pd2 {
+		pinmux = <B91_PINMUX_SET(B91_PORT_D, B91_PIN_2, B91_FUNC_A)>;
+	};
+
+	uart0_rx_pa4: uart0_rx_pa4 {
+		pinmux = <B91_PINMUX_SET(B91_PORT_A, B91_PIN_4, B91_FUNC_B)>;
+	};
+	uart0_rx_pb3: uart0_rx_pb3 {
+		pinmux = <B91_PINMUX_SET(B91_PORT_B, B91_PIN_3, B91_FUNC_C)>;
+	};
+	uart0_rx_pd3: uart0_rx_pd3 {
+		pinmux = <B91_PINMUX_SET(B91_PORT_D, B91_PIN_3, B91_FUNC_A)>;
+	};
```

Comment:
how is it possible that `uart0` `tx`/`rx` have definitions for 3 different pins? If you want to provide all valid combinations, as for example STM32 does, please create a file with valid pin combinations per SoC package instead. You may also want to check the grouping approach, that is, provide something like this instead:

```c
/* provided, may be autogenerated */
#define UART0_RX_PD3 B91_PINMUX_SET(B91_PORT_D, B91_PIN_3, B91_FUNC_A)
```

[GD32](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/gd32f450i_eval/gd32f450i_eval-pinctrl.dtsi) or [RPi Pico](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/rpi_pico/rpi_pico-pinctrl.dtsi) use this model.

I'd also suggest thinking about how you're going to support pin low-power states, since you may find that one model fits better than the other.

Reply:
Fixed. Left only the definitions which are used by the board.